tests/Feature/QuizTest.php

```diff
@@ -42,3 +36,178 @@
     expect($quiz->questions->contains($firstQuestion))->toBeTrue();
     expect($quiz->questions->contains($secondQuestion))->toBeTrue();
 });
+
+test('users can view quiz results on finishing it', function () {
+    $user = User::factory()->create();
+    $course = Course::factory()->create();
+    $user->courses()->attach(
+        $course->id, ['started_at' => now()]
+    );
+    $quiz = Quiz::factory()->for($course)->create();
+    $firstQuestion = Question::factory()->for($quiz)->create();
+    $firstQuestionCorrectChoice = Choice::factory()->for($firstQuestion)->create(['is_answer' => true]);
+    $firstQuestionWrongChoice = Choice::factory()->for($firstQuestion)->create();
+
+    $secondQuestion = Question::factory()->for($quiz)->create();
+    $secondQuestionCorrectChoice = Choice::factory()->for($secondQuestion)->create(['is_answer' => true]);
+    $secondQuestionWrongChoice = Choice::factory()->for($secondQuestion)->create();
+
+    $response = $this->actingAs($user)->get(localized_route('quizzes.show', $course));
+    $response->assertOk();
+    $response->assertSee($firstQuestion->title);
+    $response->assertSee($firstQuestionCorrectChoice->label);
+    $response->assertSee($firstQuestionWrongChoice->label);
+    $response->assertSee($secondQuestion->title);
+    $response->assertSee($secondQuestionCorrectChoice->label);
+    $response->assertSee($secondQuestionWrongChoice->label);
```

Comment:
You may want to use [`assertSeeInOrder`](https://laravel.com/docs/9.x/http-tests#assert-see-in-order) instead.